peregrine/resources/submission/graphql/node.py

```diff
@@ -433,34 +433,43 @@ def resolve_node(self, info, **args):
             for node in psqlgraph.Node.get_subclasses()
             if node.label in subclasses_labels
         ]
+        q_all = query_with_args(subclasses, args, info)
     else:
-        subclasses = [psqlgraph.Node]
-
-    q_all = []
-    for subclass in subclasses:
-        q = query_with_args(subclass, args, info)
-        q_all.extend(q.all())
+        q_all = query_node_with_args(args, info)
 
     return [__gql_object_classes[n.label](**load_node(n, info, Node.fields_depend_on_columns)) for n in q_all]
 
 
-def query_with_args(cls, args, info):
+def query_with_args(classes, args, info):
     """
-    Sends a query and applies the arguments.
+    Run queries with arguments.
 
     Args:
-        cls: psqlgraph class to query.
+        classes: psqlgraph classes to query.
         args: dictionary of the arguments passed to the query.
         info: graphene object that holds the query's arguments, models and requested fields.
     """
+    of_types = [psqlgraph.Node.get_subclass(label)
+                for label in set(args.get('of_type', []))]
+    rv = []
+    for cls in classes:
+        if not of_types or cls in of_types:
+            q = get_authorized_query(cls)
+            if 'project_id' in args:
+                q = q.filter(q.entity()._props['project_id'].astext
+                             == args['project_id'])
```

Comment:
i know this filter is in the current code, but `apply_query_args` is supposed to filter by `project_id` too... so do we need to filter again here? https://github.com/uc-cdis/peregrine/blob/b0d30160d1c607d4f9147b12b9b6d7072a9bb874/peregrine/resources/submission/graphql/node.py#L331-L334

Reply:
Oh I understand now why it is repeated:

https://github.com/uc-cdis/peregrine/blob/069066e14a27cfbb20ddb9652f76f87910929823/peregrine/resources/submission/graphql/node.py#L53

Because this one is only for the Node `project` which doesn't have a `project_id` field at all, while the other is for other nodes with `project_id` field.